examples/pebbleui/js/modules/PebbleUI.js

```diff
@@ -0,0 +1,561 @@
+var PebbleUI = function(rocky, options) {
+  options = options || {};
+
+  this._windows = [];
+
+  // Load the default font
+  this._defaultFont = options.defaultFont || rocky.FONT_KEY_GOTHIC_18;
+
+  this._getTopWindow = function() {
+    // short circuit if there's nothing to _render
+    if (!this._windows || this._windows.length < 1) return null;
+
+    // otherwise _render the active screen
+    return this._windows[this._windows.length - 1];
+  };
+
+  this._dirty = false;
+
+  this.mark_dirty = function() {
+    // don't do anything if it's already marked dirty
+    if (this._dirty === true) return;
+
+    // Otherwise, render when possible
+    this._dirty = true;
+    setTimeout(function() {
+      rocky.mark_dirty();
+      this._dirty = false;
+    }.bind(this), 0);
+  };
+
+  this._render = function(ctx, bounds) {
+    var win = this._getTopWindow();
+    if (win) win._render(ctx, bounds);
+  };
+
+  var parent = this;
+
+  this._window = window || {};
+  this._keys = {};
+
+  this._window.addEventListener('keydown', function(event) {
+    var code = event.keyCode;
+    var keys = parent._keys;
+    var win = parent._getTopWindow();
+    if (!win) return;
+
+    if (code < 37 || code > 40) return;   // ignore everything except the arrow keys
+    event.preventDefault();
+
+    if (!keys[code]) {
+      keys[code] = { timedown: event.timeStamp };
+      win._invokeButtonCallbackByCode(code, { type: 'short' });
+    }
+
+    if (event.timeStamp - keys[code].timedown > 500 && !keys[code].longPress) {
+      keys[code].longPress = true;  // Mark that we've fired the longPress
+      win._invokeButtonCallbackByCode(code, { type: 'long' });
+    }
+  });
+
+  this._window.addEventListener('keyup', function(event) {
+    var code = event.keyCode;
+    var keys = parent._keys;
+
+    if (code < 37 || code > 40) return;   // ignore everything except the arrow keys
+
+    delete keys[code];
+  });
+
+  // Everything is an Element!
+  this.Element = function(options) {
```

Comment:
as you're describing a traditional type hierarchy, do you think it makes sense to use prototypes?